packages/framework/src/Model/Order/OrderDeliveryDateFacade.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\Order;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Shopsys\FrameworkBundle\Component\DateTimeHelper\DateTimeHelper;
+use Shopsys\FrameworkBundle\Component\Localization\DisplayTimeZoneProviderInterface;
+use Shopsys\FrameworkBundle\Model\Order\Status\OrderStatusTypeEnum;
+
+class OrderDeliveryDateFacade
+{
+    /**
+     * @param \Doctrine\ORM\EntityManagerInterface $em
+     * @param \Shopsys\FrameworkBundle\Component\Localization\DisplayTimeZoneProviderInterface $displayTimeZoneProvider
+     * @param \Shopsys\FrameworkBundle\Component\DateTimeHelper\DateTimeHelper $dateTimeHelper
+     */
+    public function __construct(
+        protected readonly EntityManagerInterface $em,
+        protected readonly DisplayTimeZoneProviderInterface $displayTimeZoneProvider,
+        protected readonly DateTimeHelper $dateTimeHelper,
+    ) {
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Order\Order $order
+     */
+    public function setDeliveredNowIfNecessary(Order $order): void
+    {
+        if ($this->shouldSetDeliveredAt($order)) {
+            $this->setDeliveredAt($order);
+            $this->em->flush();
+        }
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Order\Order $order
+     * @return bool
+     */
+    protected function shouldSetDeliveredAt(Order $order): bool
+    {
+        return $order->getDeliveredAt() === null && ($order->getStatus()->getType() === OrderStatusTypeEnum::TYPE_DONE);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Order\Order $order
+     */
+    protected function setDeliveredAt(Order $order): void
+    {
+        $domainTimeZone = $this->displayTimeZoneProvider->getDisplayTimeZoneByDomainId($order->getDomainId());
+        $deliveredAt = $this->dateTimeHelper->createUtcDateTimeByTimeZoneAndString('now', $domainTimeZone);
```

Comment:
What's the point of the conversion? isn't `new DateTime('now')` enough?

Reply:
eh, fixed tests in 304ee3eb6e